lib/gui/pages/main/controllers/flash.js

```diff
@@ -16,19 +16,98 @@
 
 'use strict'
 
+const _ = require('lodash')
 const messages = require('../../../../shared/messages')
 const flashState = require('../../../../shared/models/flash-state')
 const driveScanner = require('../../../modules/drive-scanner')
 const progressStatus = require('../../../modules/progress-status')
 const notification = require('../../../os/notification')
 const exceptionReporter = require('../../../modules/exception-reporter')
 const path = require('path')
+const constraints = require('../../../../shared/drive-constraints')
+const units = require('../../../../shared/units')
+const errors = require('../../../../shared/errors')
 
 module.exports = function (
+  $q,
   $state,
   ImageWriterService,
-  FlashErrorModalService
+  FlashErrorModalService,
+  WarningModalService,
+  DriveSelectorService
 ) {
+  /**
+   * @summary Spawn a confirmation warning modal
+   * @function
+   * @public
+   *
+   * @param {String} message - warning message
+   * @returns {Promise} warning modal promise
+   *
+   * @example
+   * confirmationWarningModal('Hello, World!')
+   */
+  const confirmationWarningModal = (message) => {
+    return WarningModalService.display({
+      confirmationLabel: 'Continue',
+      rejectionLabel: 'Change',
+      description: [
+        message,
+        'Are you sure you want to continue?'
+      ].join(' ')
+    })
+  }
+
+  /**
+   * @summary Display warning tailored to the warning of the current drive-image pair
+   * @function
+   * @public
+   *
+   * @param {Object} drive - drive object
+   * @param {Object} image - image object
+   * @returns {Promise}
+   *
+   * @example
+   * displayTailoredWarning(drive, image).then(() => {
+   *   console.log('Continue pressed')
+   * }).catch(() => {
+   *   console.log('Change pressed')
+   * })
+   */
+  const displayTailoredWarning = (drive, image) => {
+    const messageKeys = {
+      largeDriveSize: constraints.isDriveSizeLarge(drive),
+      unrecommendedDriveSize: !constraints.isDriveSizeRecommended(drive, image)
```

Comment:
Hmm, I'm not sure if unrecommendedDriveSize is a serious-enough-problem to warrant displaying an extra warning modal to the user?

Reply:
We could remove it but it's the current behaviour, so if we want to remove it I think we should open a separate issue.